presto-spi/src/main/java/io/prestosql/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -511,7 +515,7 @@ public static void multiply(Slice left, Slice right, Slice result)
             }
         }
 
-        pack(result, (int) z0, (int) z1, (int) z2, (int) z3, isNegative(left) != isNegative(right));
+        pack(result, (int) z0, (int) z1, (int) z2, (int) z3, leftNegative != rightNegative);
```

Comment:
there are more places where it could be optimized, e.g: `io.prestosql.spi.type.UnscaledDecimal128Arithmetic#multiplyDestructive`

Reply:
This is just the beginning.
I made a heavily-optimised version already but benchmarks went crazy. 
One step at a time